File: frontend/src/services/commentService.js
import { apiClient } from "./apiClient";

export class CommentService {

    getCommentsByMovieId(movieId, pageNo, pageSize=5) {
        return apiClient.get("/movie/comments/getCommentsByMovieId/" + movieId + "/" + pageNo + "/" + pageSize);
    }

    getCountOfComments(movieId) {
        return apiClient.get("/movie/comments/getCountOfComments/" + movieId);
    }

    addComment(commentDto) {
        return apiClient.post("/movie/comments/add" , commentDto);
    }
    
    deleteComment(deleteCommentDto) {
        return apiClient.post("/movie/comments/delete" , deleteCommentDto);
    }
}

File: movieService/src/main/java/com/doffeii/movieService/business/abstracts/CommentService.java
package com.doffeii.movieService.business.abstracts;

import com.doffeii.movieService.entity.Comment;
import com.doffeii.movieService.entity.dto.CommentRequestDto;
import com.doffeii.movieService.entity.dto.DeleteCommentRequestDto;

import java.util.List;

public interface CommentService {

    List<Comment> getCommentsByMovieId(int movieId, int pageNo, int pageSize);

    void deleteComment(DeleteCommentRequestDto deleteCommentRequestDto, String authorizationHeader);

    Comment addComment(CommentRequestDto commentRequestDto, String authorizationHeader);

    int getNumberOfCommentsByMovieId(int movieId);

    double getAverageRatingByMovieId(int movieId);
}
